app/api/internal/users.rb

```diff
@@ -44,6 +44,16 @@ def self.save(service_id, username, usr_attrs, method, headers)
             [400, headers, { status: :error, error: e.message }.to_json]
           end
         end
+
+        delete '/all' do |service_id|
```

Comment:
as @davidor suggested, maybe 

`DELETE /internal/services/:service_id/users` would be more RESTful

can be implemented by

`delete '' do |service_id|`

Reply:
:+1:

Reply:
What happens if I have a following URL constructing code:

```ruby
http.delete("/internal/services/#{service_id}/users/#{username}")
```

and `username` is nil? So the resulting URL is `/internal/services/:service_id/users/` ? Would it delete all users?

Reply:
It shouldn't be a problem.

If you call `/internal/services/:service_id/users/` then it would return a 404 not found because that URL has not been defined in the internal API.

For example, if I change the current internal api test code to execute this code (notice the slash at the end):
```
  delete '/services/:service_id/users/' do
    parameter :service_id, 'Service ID', required: true

    example_request 'Deleting all Users of a Service' do
      expect(status).to eq 200
      expect(response_json['status']).to eq 'deleted'
      expect(ThreeScale::Backend::User.exists?(service_id, username)).to be false
    end
  end
```

and I keep the internal api code as:
```
        delete '' do |service_id|
          begin
            User.delete_all(service_id)
            { status: :deleted}.to_json
          rescue => e
            [400, headers, { status: :error, error: e.message }.to_json]
          end
        end
```

When I execute the test I get the following:
```
Failures:

  1) Users (prefix: /services/:service_id/users) DELETE /services/:service_id/users/ Deleting all Users of a Service
     Failure/Error: expect(status).to eq 200
     
       expected: 200
            got: 404
     
       (compared using ==)
     # ./spec/acceptance/api/internal/users_api_spec.rb:151:in `block (3 levels) in <top (required)>'

Finished in 0.90085 seconds (files took 0.68985 seconds to load)
144 examples, 1 failure

Failed examples:

rspec ./spec/acceptance/api/internal/users_api_spec.rb:150 # Users (prefix: /services/:service_id/users) DELETE /services/:service_id/users/ Deleting all Users of a Service
```

Reply:
👍